lib/app/index.js

```diff
@@ -1,53 +1,171 @@
-var async = require('async');
+'use strict';
 
-var mage = require('../mage');
-var logger = mage.core.logger;
+const memoize = require('memoizee');
 
 
 // app registration
 
-var appMap = {};
-var appList = [];
+const appMap = {};
+const appList = [];
 
 
 /**
- * This function creates instances of all configured apps and registers them into appMap and appList.
+ * Returns the configuration for all apps (overridden during unit tests)
+ *
+ * @returns {Object}
+ */
+/* istanbul ignore next */
+exports.getAppsConfig = memoize(function () {
+	return require('lib/mage').core.config.get(['apps']);
```

Comment:
A require call is still relatively expensive. Paths have to be resolved for example to find the registered (or not) module. Memoizing this stuff just makes this much cheaper to call.